File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] heights) {

        int p = 0, q = heights.length-1;
        int max = 0;
        int area = 0;

        while(p<q){
            if(heights[p]<heights[q]){
                area = heights[p]*(q-p);
                p++;
            }else{
                area = heights[q]*(q-p);
                q--;
            }
            if(area>=max) max = area;
        }

        return max;
        
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        n = len(height)
        s, e = 0,n-1
        maxVol = 0

        while(s<e):
            maxVol = max(maxVol, (e-s) * min(height[e], height[s]))
            if(height[e]>height[s]):
                s += 1
            else:
                e -= 1
        return maxVol